File: dao/CameraDAO.php
<?php

class CameraDAO {
    private $pdo;
    
    public function __construct($pdo) {
        $this->pdo = $pdo;
    }
    
    // Obtém o ID da câmera
    public function getCameraId($modeloEditar, $MACEditar) {
        $sql = "SELECT COUNT(*) AS total FROM cameras WHERE modelo = :modelo AND MAC = :MAC";
        $cameraId = 0;

        try {
            $stmt = $this->pdo->prepare($sql);
            $stmt->execute(['modelo' => $modeloEditar, 'MAC' => $MACEditar]);

            $totalFerramentas = $stmt->fetchColumn();
            if ($totalFerramentas > 0) {
                $sql = "SELECT cameraid FROM cameras WHERE modelo = :modelo AND MAC = :MAC";
                $stmt2 = $this->pdo->prepare($sql);
                $stmt2->execute(['modelo' => $modeloEditar, 'MAC' => $MACEditar]);
                $cameraId = $stmt2->fetchColumn();
            } else {
                return -1;
            }
        } catch (PDOException $e) {
            echo "Erro ao procurar câmera: " . $e->getMessage();
        }

        return $cameraId;
    }

    // Obtém o modelo da câmera pelo ID
    public function getModelo($cameraId) {
        $sql = "SELECT modelo FROM cameras WHERE cameraid = :cameraid";
        $modelo = "";

        try {
            $stmt = $this->pdo->prepare($sql);
            $stmt->execute(['cameraid' => $cameraId]);
            $modelo = $stmt->fetchColumn();

            if ($modelo === false) {
                echo "Nenhum modelo encontrado com o id: " . $cameraId;
            }
        } catch (PDOException $e) {
            echo "Erro ao buscar o modelo da câmera: " . $e->getMessage();
        }

        return $modelo;
    }

    // Obtém o MAC da câmera pelo ID
    public function getMAC($cameraId) {
        $sql = "SELECT MAC FROM cameras WHERE cameraid = :cameraid";
        $MAC = "";

        try {
            $stmt = $this->pdo->prepare($sql);
            $stmt->execute(['cameraid' => $cameraId]);
            $MAC = $stmt->fetchColumn();

            if ($MAC === false) {
                echo "Nenhum MAC encontrado com o id: " . $cameraId;
            }
        } catch (PDOException $e) {
            echo "Erro ao buscar o MAC da câmera: " . $e->getMessage();
        }

        return $MAC;
    }

    // Atualiza o modelo da câmera
    public function setModelo($cameraId, $novoModelo) {
        $sql = "UPDATE cameras SET modelo = :modelo WHERE cameraid = :cameraid";

        try {
            $stmt = $this->pdo->prepare($sql);
            $stmt->execute(['modelo' => $novoModelo, 'cameraid' => $cameraId]);
        } catch (PDOException $e) {
            echo "Erro: " . $e->getMessage();
        }
    }

    // Atualiza o MAC da câmera
    public function setMAC($cameraId, $novoMAC) {
        $sql = "UPDATE cameras SET MAC = :MAC WHERE cameraid = :cameraid";

        try {
            $stmt = $this->pdo->prepare($sql);
            $stmt->execute(['MAC' => $novoMAC, 'cameraid' => $cameraId]);
        } catch (PDOException $e) {
            echo "Erro: " . $e->getMessage();
        }
    }

    // Retorna a lista de câmeras
    public function getCameras() {
        $minhaLista = [];

        try {
            $sql = "SELECT * FROM cameras";
            $stmt = $this->pdo->query($sql);

            while ($row = $stmt->fetch(PDO::FETCH_ASSOC)) {
                $nome = $row['modelo'];
                $MAC = $row['MAC'];
                $camera = new Camera($nome, $MAC); // Assumindo que a classe Camera já existe
                $minhaLista[] = $camera;
            }
        } catch (PDOException $e) {
            echo "Erro: " . $e->getMessage();
        }

        return $minhaLista;
    }

    // Adiciona uma nova câmera
    public function addCamera($modelo, $MAC) {
        $sql = "INSERT INTO cameras (modelo, MAC) VALUES (:modelo, :MAC)";

        try {
            $stmt = $this->pdo->prepare($sql);
            $stmt->execute(['modelo' => $modelo, 'MAC' => $MAC]);
        } catch (PDOException $e) {
            echo "Erro: " . $e->getMessage();
        }
    }

    // Exclui uma câmera
    public function delCamera($cameraId) {
        $sql = "DELETE FROM cameras WHERE cameraid = :cameraid";

        try {
            $stmt = $this->pdo->prepare($sql);
            $stmt->execute(['cameraid' => $cameraId]);
        } catch (PDOException $e) {
            echo "Erro: " . $e->getMessage();
        }
    }
}

// Exemplo de uso
try {
    $pdo = new PDO("mysql:host=10.100.68.85;dbname=envio;charset=utf8", "root", "Camerasip135.");
    $pdo->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);

    $cameraDAO = new CameraDAO($pdo);
    
    // Exemplo: adicionar uma nova câmera
    $cameraDAO->addCamera("Modelo X", "00:14:22:01:23:45");

    // Exemplo: obter a lista de câmeras
    $cameras = $cameraDAO->getCameras();
    foreach ($cameras as $camera) {
        echo "Modelo: " . $camera->getModelo() . ", MAC: " . $camera->getMAC() . "<br>";
    }
} catch (PDOException $e) {
    echo "Erro na conexão: " . $e->getMessage();
}

?>

File: dao/CameraDAO.py
import sqlite3
import mysql.connector

class CameraDAO:
    def __init__(self, modelo="", mac=""):
        self.modelo = modelo
        self.mac = mac
        self.connection = self.conectar()

    def conectar(self):
        banco = "envio.db"  # Ajuste o caminho do banco de dados conforme necessário
        try:
            connection = sqlite3.connect(banco)
            return connection
        except sqlite3.Error as e:
            print(f"Erro ao conectar ao banco de dados: {e}")
            return None
        
    # Obtém o ID da câmera
    def get_camera_id(self, modelo_editar, mac_editar):
        sql = "SELECT COUNT(*) AS total FROM cameras WHERE modelo = ? AND MAC = ?"
        camera_id = 0
        
        try:
            cursor = self.connection.cursor()
            cursor.execute(sql, (modelo_editar, mac_editar))
            total_ferramentas = cursor.fetchone()[0]

            if total_ferramentas > 0:
                sql = "SELECT cameraid FROM cameras WHERE modelo = ? AND MAC = ?"
                cursor.execute(sql, (modelo_editar, mac_editar))
                camera_id = cursor.fetchone()[0]
            else:
                return -1
        except sqlite3.Error as e:
            print(f"Erro ao procurar câmera: {e}")
        
        return camera_id

    # Obtém o modelo da câmera pelo ID
    def get_modelo_dao(self, camera_id):
        db_config = {
            'user': 'root',
            'password': "Camerasip135.",
            'host': "10.100.68.253",
            'database': "envio"
        }
        try: 
            conn = mysql.connector.connect(**db_config)
            cursor = conn.cursor(dictionary=True)
            sql = "SELECT modelo FROM cameras WHERE cameraid = %s"
            cursor.execute(sql, (camera_id,))
            modelo = cursor.fetchone()
            cursor.close()
            conn.close()
            
            if modelo:
                return modelo['modelo']
            else:
                print(f"Nenhum modelo encontrado com o id: {camera_id}")
                return ""
        except mysql.connector.Error as e:
            print(f"Erro ao buscar o modelo da câmera: {e}")
            return ""

    # Obtém o MAC da câmera pelo ID
    def get_mac_dao(self, camera_id):
        sql = "SELECT MAC FROM cameras WHERE cameraid = ?"
        mac = ""

        try:
            cursor = self.connection.cursor()
            cursor.execute(sql, (camera_id,))
            mac = cursor.fetchone()
            if mac:
                mac = mac[0]
            else:
                print(f"Nenhum MAC encontrado com o id: {camera_id}")
        except sqlite3.Error as e:
            print(f"Erro ao buscar o MAC da câmera: {e}")

        return mac

    # Atualiza o modelo da câmera
    def set_modelo(self, camera_id, novo_modelo):
        sql = "UPDATE cameras SET modelo = ? WHERE cameraid = ?"

        try:
            cursor = self.connection.cursor()
            cursor.execute(sql, (novo_modelo, camera_id))
            self.connection.commit()
        except sqlite3.Error as e:
            print(f"Erro: {e}")

    # Atualiza o MAC da câmera
    def set_mac(self, camera_id, novo_mac):
        sql = "UPDATE cameras SET MAC = ? WHERE cameraid = ?"

        try:
            cursor = self.connection.cursor()
            cursor.execute(sql, (novo_mac, camera_id))
            self.connection.commit()
        except sqlite3.Error as e:
            print(f"Erro: {e}")

    # Retorna a lista de câmeras
    def get_cameras(self):
        minha_lista = []

        try:
            sql = "SELECT * FROM cameras"
            cursor = self.connection.cursor()
            cursor.execute(sql)

            rows = cursor.fetchall()
            for row in rows:
                nome = row[1]  # Considerando que o modelo está na segunda coluna
                mac = row[2]   # Considerando que o MAC está na terceira coluna
                camera = Camera(nome, mac)  # Assumindo que a classe Camera já existe
                minha_lista.append(camera)
        except sqlite3.Error as e:
            print(f"Erro: {e}")

        return minha_lista

    # Adiciona uma nova câmera
    def add_camera(self, modelo, mac):
        sql = "INSERT INTO cameras (modelo, MAC) VALUES (?, ?)"

        try:
            cursor = self.connection.cursor()
            cursor.execute(sql, (modelo, mac))
            self.connection.commit()
        except sqlite3.Error as e:
            print(f"Erro: {e}")

    # Exclui uma câmera
    def del_camera(self, camera_id):
        sql = "DELETE FROM cameras WHERE cameraid = ?"

        try:
            cursor = self.connection.cursor()
            cursor.execute(sql, (camera_id,))
            self.connection.commit()
        except sqlite3.Error as e:
            print(f"Erro: {e}")

# Exemplo de uso
def main():
    try:
        connection = sqlite3.connect("envio.db")  # Ajuste o caminho do banco de dados conforme necessário
        camera_dao = CameraDAO(connection)
        
        # Exemplo: adicionar uma nova câmera
        camera_dao.add_camera("Modelo X", "00:14:22:01:23:45")

        # Exemplo: obter a lista de câmeras
        cameras = camera_dao.get_cameras()
        for camera in cameras:
            print(f"Modelo: {camera.get_modelo()}, MAC: {camera.get_mac()}")

    except sqlite3.Error as e:
        print(f"Erro na conexão: {e}")

if __name__ == "__main__":
    main()
